specifyweb/frontend/js_src/lib/components/Attachments/RecordSetAttachment.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+
+import { useAsyncState } from '../../hooks/useAsyncState';
+import { useCachedState } from '../../hooks/useCachedState';
+import { attachmentsText } from '../../localization/attachments';
+import { commonText } from '../../localization/common';
+import type { RA } from '../../utils/types';
+import { filterArray } from '../../utils/types';
+import { Button } from '../Atoms/Button';
+import { serializeResource } from '../DataModel/helpers';
+import type { AnySchema } from '../DataModel/helperTypes';
+import type { SpecifyResource } from '../DataModel/legacyTypes';
+import type { CollectionObjectAttachment } from '../DataModel/types';
+import { Dialog } from '../Molecules/Dialog';
+import { defaultScale } from '.';
+import { AttachmentGallery } from './Gallery';
+
+export function RecordSetAttachments<SCHEMA extends AnySchema>({
+  records,
+  onClose: handleClose,
+  onFetch: handleFetch,
+}: {
+  readonly records: RA<SpecifyResource<SCHEMA> | undefined>;
+  readonly onClose: () => void;
+  readonly onFetch?:
+    | ((index: number) => Promise<RA<number | undefined> | void>)
+    | undefined;
+}): JSX.Element {
+  const recordFetched = React.useRef<number>(0);
```

Comment:
`recordFetched` variable names makes it sounds like it either holds the fetched records, or a boolean indicating whether a record was fetched.

a better name would be `fetchedCount`